### PR TITLE
 Add reader for OFF file in paraview

### PR DESCRIPTION
**Pull request by CharlesGueunet**
Dear Julien,

This pull request add a reader in Paraview to support Object File Format files.
With this reader, Paraview can open a mesh in .off and generate the corresponding VTU.

Charles

### CODE REVIEW

**Comment by julien-tierny** [comment]
hi Charles,
thanks for the initiative, this is great!
the travis build failed though. 
it seems a file is missing an #include <vtkIdList.h>. can you fix that real quick?
thanks

**Comment by julien-tierny** [comment]
I also added a comment in the code that may require a fix. Thanks.

File: core/vtk/ttkOFFReader/ttkOFFReader.cpp [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  vtkSmartPointer<vtkPoints> points = vtkSmartPointer<vtkPoints>::New();
+  vtkSmartPointer<vtkUnstructuredGrid> mesh =
+      vtkSmartPointer<vtkUnstructuredGrid>::New();
+  vtkSmartPointer<vtkDoubleArray> scalars =
+      vtkSmartPointer<vtkDoubleArray>::New();
+
+  int nbVerts, nbFaces, nbEdges, curLine;
+  std::string line;
+
+  offFile >> nbVerts >> nbFaces >> nbEdges;
```
Comment (julien-tierny) [inline core/vtk/ttkOFFReader/ttkOFFReader.cpp]: I don't believe most OFF files provide the number of edges in the header (look at examples on the AIM@SHAPE repository for instance)

Comment (CharlesGueunet) [reply core/vtk/ttkOFFReader/ttkOFFReader.cpp]: This is fixed in commit 2dffc5a53fa9b52fa564ff0e874c7f6f6e0cf0e7

File: core/vtk/ttkOFFReader/ttkOFFReader.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  while (curLine < nbVerts && std::getline(offFile, line)) {
+    double x, y, z, w;
+    std::stringstream ss(line);
+    if (!(ss >> x >> y >> z >> w)) {
+      // if the scalar is not on the file, we use an elevation on the x axis.
+      w = x;
+    }
+
+    points->InsertNextPoint(x, y, z);
+    scalars->SetTuple1(curLine, w);
```
[the remark this answers is not in the document]
Comment (CharlesGueunet) [reply core/vtk/ttkOFFReader/ttkOFFReader.cpp]: This is fixed in commit a7408065c7de02208b3d059bf577db582b5f8844

**Comment by julien-tierny** [comment]
That's just awesome.
Thanks.

**Pull request merged**